messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -102,17 +107,16 @@ internal class GistSdk(
         }
 
         inAppMessagingManager.subscribeToAttribute({ it.pollInterval }) { interval ->
+            val currentState = state
+            if (currentState.sseEnabled) {
+                return@subscribeToAttribute
+            }
             fetchInAppMessages(duration = interval, initialDelay = interval)
         }
 
-        // Subscribe to SSE flag changes for dynamic switching
         inAppMessagingManager.subscribeToAttribute({ it.sseEnabled }) { sseEnabled ->
-            // TODO ensure this respects lifecycle foreground/background
-            logger.info("SSE flag changed to: $sseEnabled")
             if (sseEnabled) {
```

Comment:
what happens in the negative case? what if its disabled, how do we switch back to polling?